src/tests/Regressions/coreclr/GitHub_115667/test115667.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using Xunit;
+using System.Threading.Tasks;
+
+public interface I0
+{
+    Task<bool> M8();
+}
+
+public struct S1 : I0
+{
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    public async Task<bool> M8()
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+    {
+        return false;
+    }
+}
+
+public class Program
+{
+    [Fact]
+    public static int TestEntryPoint()
+    {
+        System.Runtime.Loader.AssemblyLoadContext alc = new CollectibleALC();
+        System.Reflection.Assembly asm = alc.LoadFromAssemblyPath(System.Reflection.Assembly.GetExecutingAssembly().Location);
+        System.Reflection.MethodInfo mi = asm.GetType(typeof(Program).FullName).GetMethod(nameof(MainInner));
+        System.Type runtimeTy = asm.GetType(typeof(Runtime).FullName);
+        mi.Invoke(null, new object[] { System.Activator.CreateInstance(runtimeTy) });
+
+        return 100;
```

Comment:
This is unnecessary (and will fail a build analyzer we have in some builds). The test can return void instead.